NBXplorer.Client/NBXplorerNetworkProvider.Liquid.cs

```diff
@@ -0,0 +1,209 @@
+﻿using NBitcoin;
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using NBitcoin.Altcoins.Elements;
+using NBitcoin.JsonConverters;
+using NBitcoin.RPC;
+using NBXplorer.DerivationStrategy;
+using NBXplorer.Models;
+using Newtonsoft.Json;
+
+namespace NBXplorer
+{
+	public partial class NBXplorerNetworkProvider
+	{
+
+		public class LiquidExplorerClient : ExplorerClient
+		{
+			public LiquidExplorerClient(NBXplorerNetwork network, Uri serverAddress = null) : base(network, serverAddress)
+			{
+			}
+			public override async Task<TransactionResult> GetTransactionAsync(uint256 txId,
+				CancellationToken cancellation = default)
+			{
+				return await base.GetTransactionAsync<LiquidTransactionResult>(txId, cancellation);
+			}
+		}
+
+
+		public class  LiquidTransactionResult : TransactionResult
+		{
+			private ElementsTransaction _transaction;
+			public override Transaction Transaction
+			{
+				get => _transaction;
+				set => _transaction = value as ElementsTransaction;
+			}
+		}
+		
+		public class LiquidNBXplorerNetwork : NBXplorerNetwork
+		{
+			public LiquidNBXplorerNetwork(INetworkSet networkSet, NetworkType networkType, DerivationStrategyFactory derivationStrategyFactory = null) : base(networkSet, networkType, derivationStrategyFactory)
+			{
+			}
+
+			public override ExplorerClient CreateExplorerClient(Uri uri)
+			{
+				return new LiquidExplorerClient(this, uri);
+			}
+			
+			public override BitcoinAddress CreateAddress(DerivationStrategyBase derivationStrategy, KeyPath keyPath, Script scriptPubKey)
+			{
+				if (derivationStrategy is IHasDerivationStrategyOptions strategyOptions &&
+				    strategyOptions.DerivationStrategyOptions.AdditionalOptions.TryGetValue("unblinded",
+					    out var unblinded) && (bool) unblinded)
+				{
+					return base.CreateAddress(derivationStrategy, keyPath, scriptPubKey);
+				}
+				
+				var blindingKey = GenerateBlindingKey(derivationStrategy, keyPath);
+				if (blindingKey == null)
+				{
+					return base.CreateAddress(derivationStrategy, keyPath, scriptPubKey);
+				}
+				var blindingPubKey = blindingKey.PubKey;
+				return new BitcoinBlindedAddress(blindingPubKey, base.CreateAddress(derivationStrategy, keyPath, scriptPubKey));
+			}
+
+			public static Key GenerateBlindingKey(DerivationStrategyBase derivationStrategy, KeyPath keyPath)
+			{
+				if (derivationStrategy == null)
```

Comment:
why derivationStrategy would be null?

Reply:
There is one place at least: 
```
public async Task<MultiValueDictionary<Script, KeyPathInformation>> GetKeyInformations(Script[] scripts)
		{
			MultiValueDictionary<Script, KeyPathInformation> result = new MultiValueDictionary<Script, KeyPathInformation>();
			if (scripts.Length == 0)
				return result;
			foreach (var batch in scripts.Batch(BatchSize))
			{
				await _TxContext.DoAsync(tx =>
				{
					tx.ValuesLazyLoadingIsOn = false;
					foreach (var script in batch)
					{
						var table = GetScriptsIndex(tx, script);
						var keyInfos = table.SelectForwardSkip(0)
											.Select(r => GetKeyPathInformation(r.Value, null))
											// Because xpub are mutable (several xpub map to same script)
											// an attacker could generate lot's of xpub mapping to the same script
											// and this would blow up here. This we take only 5 results max.
											.Take(5)
											.ToArray();
						result.AddRange(script, keyInfos);
					}
				});
			}
			return result;
		}


		protected virtual KeyPathInformation GetKeyPathInformation(byte[] value, DerivationStrategyBase derivationStrategyBase)
		{
			var result = ToObject<KeyPathInformation>(value);
			if (result.Address == null)
			{
				result.Address = Network.CreateAddress(derivationStrategyBase, result.KeyPath, result.ScriptPubKey);
			}

			return result;
		}
```

Reply:
Before there was `AddAddress` with my change `AddAddress` will only apply to old KeyInfo.
New KeyInfo have their address saved in DB.

So this should never happen.

Reply:
Ok,